addons/sourcemod/scripting/shavit-tas.sp

```diff
@@ -0,0 +1,1407 @@
+	Shavit_PrintToChat(client, "!tasmenu - Toggles TAS Menu");
+	return Plugin_Handled;
+}
+
+public Action CommandListener_TAS(int client, const char[] command, int args)
+{
+	gB_TASMenu[client] = true;
+	DrawPanel(client);
+	return Plugin_Continue;
+}
+
+public Action CommandListener_PlusRewind(int client, const char[] command, int args)
+{
+	if(!gB_TAS[client])
+	{
+		return Plugin_Handled;
+	}
+	gI_Status[client] = BACKWARD;
+	return Plugin_Handled;
+}
+
+public Action CommandListener_PlusForward(int client, const char[] command, int args)
+{
+	if(!gB_TAS[client])
+	{
+		return Plugin_Handled;
+	}
+	gI_Status[client] = FORWARD;
+	return Plugin_Handled;
+}
+
+public Action CommandListener_MinusRewindOrForward(int client, const char[] command, int args)
+{
+	if(!gB_TAS[client])
+	{
+		return Plugin_Handled;
+	}
+	gI_Status[client] = PAUSED;
+	return Plugin_Handled;
+}
+
+public Action OnPlayerRunCmd(int client, int &buttons, int &impulse, float vel[3], float angles[3], int &weapon, int &subtype, int &cmdnum, int &tickcount, int &seed, int mouse[2])
+{
+	if(gB_TAS[client] && IsValidClient(client, true))
+	{
+		DrawPanel(client);
+
+		if(Shavit_GetTimerStatus(client) != Timer_Running)
+		{
+			return Plugin_Continue;
+		}
+		else
+		{
+			if(gI_Status[client] == RUN)
+			{
+				// Record Frames
+				//SetEntPropFloat(client, Prop_Send, "m_flLaggedMovementValue", gF_Timescale[client]);
+				float fTimeScale = GetEntPropFloat(client, Prop_Data, "m_flLaggedMovementValue");
+				gF_TimescaleTicksPassed[client] += fTimeScale;
+
+				float fDifference = angles[1] - gF_LastAngle[client];
+				if (fDifference > 180.0)
+				{
+					fDifference -= 360.0;
+				}
+				else if(fDifference < -180.0)
+				{
+					fDifference += 360.0;
+				}
+				/*
+							AUTO STRAFER START
+														*/
+				if((buttons & IN_FORWARD) > 0 && vel[0] <= 50.0)
+				{
+					vel[0] = gF_MaxMove;
+				}
+
+				float fYawChange = 0.0;
+				if(vel[0] > 50.0)
+				{
+					fYawChange = gF_AirSpeedCap * FloatAbs(gF_AirSpeedCap / vel[0]);
+				}
+
+				if (gB_AutoStrafeEnabled[client] && (GetEntityFlags(client) & FL_ONGROUND) == 0 && (GetEntityMoveType(client) != MOVETYPE_NOCLIP)
+					&& (buttons & IN_FORWARD) == 0 && (buttons & IN_BACK) == 0 && (buttons & IN_MOVELEFT) == 0 && (buttons & IN_MOVERIGHT) == 0)
+				{
+					if(fDifference < 0.0)
+					{
+						angles[1] += fYawChange;
+						//buttons |= IN_MOVERIGHT;
+						vel[1] = gF_MaxMove;
+					}
+					else if(fDifference > 0.0)
+					{
+						angles[1] -= fYawChange;
+						//buttons |= IN_MOVELEFT;
+						vel[1] = -gF_MaxMove;
+					}
+				}
+				/*
+							AUTO STRAFER END
+														*/
+
+				/*
+						WIGGLEHACK START
+						Huge amounts of credit to Kamay for this code!
+						https://steamcommunity.com/id/xutaxkamay/
+																		*/
+				if (gB_SilentStrafe[client] && (GetEntityFlags(client) & FL_ONGROUND) == 0 && (GetEntityMoveType(client) != MOVETYPE_NOCLIP)
```

Comment:
if statement could be merged with line 383 by checking the flags, movetype, and buttons first then inside that if statement check the bool.